starlite/openapi/parameters.py

```diff
@@ -65,66 +72,127 @@ def list(self) -> List[Parameter]:
-            extra = field.field_info.extra
-            header_key = extra.get("header")
-            cookie_key = extra.get("cookie")
-            query_key = extra.get("query")
-            if f_name in path_parameter_names:
-                param_in = "path"
-                required = True
-                schema = create_path_parameter_schema(
-                    path_parameter=[p for p in path_parameters if f_name in p["name"]][0],
-                    field=field,
-                    generate_examples=generate_examples,
-                )
-            elif header_key:
-                f_name = header_key
-                param_in = "header"
-                required = field.field_info.extra["required"]
-            elif cookie_key:
-                f_name = cookie_key
-                param_in = "cookie"
-                required = field.field_info.extra["required"]
-            elif query_key:
-                f_name = query_key
-                param_in = "query"
-                required = cast(bool, field.required) if field.required is not Undefined else False
-            if not schema:
-                schema = create_schema(field=field, generate_examples=generate_examples)
+        else:
             parameters.add(
-                Parameter(
-                    name=f_name,
-                    param_in=param_in,
-                    required=required,
-                    param_schema=schema,
-                    description=schema.description,
+                create_parameter(
+                    model_field=model_field,
+                    parameter_name=field_name,
+                    path_parameters=path_parameters,
+                    generate_examples=generate_examples,
                 )
             )
+    for field_name, model_field in filter(
+        lambda items: items[0] not in RESERVED_KWARGS and items[0] not in handler_fields, layered_parameters.items()
+    ):
+        parameters.add(
+            create_parameter(
+                model_field=model_field,
+                parameter_name=field_name,
+                path_parameters=path_parameters,
+                generate_examples=generate_examples,
+            )
+        )
+    for field_name, signature_model_filed in filter(
+        lambda items: items[0] not in RESERVED_KWARGS and items[0] in layered_parameters, handler_fields.items()
+    ):
+
+        layer_field_info = layered_parameters[field_name].field_info
+        signature_field_info = signature_model_filed.field_info
+
+        field_info = layer_field_info
+        # allow users to manually override Parameter definition using Parameter
+        if signature_field_info.extra.get(EXTRA_KEY_IS_PARAMETER):
+            field_info = signature_field_info
+
+        field_info.default = (
+            signature_field_info.default
+            if signature_field_info.default not in [Undefined, Ellipsis]
+            else layer_field_info.default
+        )
```

Comment:
```python

class MyController(Controller):

    parameters = {"param": Parameter(value_type=int, default=5)}
    route = "/"

    @get("override-default")
    def override_default(self, param: int = Parameter(default=...)) -> dict[str, Any]:
        """GET /with-default"""
        # returns {"param": 5}
        return {"param": param}
```
Just questioning whether `Parameter(default=...)` should be treated different to overriding the parameter with something like `Parameter(default=6)`? I think it would have less WTF factor if the ellipsis thing worked.

Reply:
So, what should default ellipsis do?